tests/test_template.py

```diff
@@ -69,37 +74,51 @@ def test_load_environment_vars_with_prefix_with_trailing_underscore(monkeypatch)
     assert values['SHPKPR_APPLE_AND_BLACKCURRANT'] == 'crumble'
 
 
-def test_render_json_template_valid():
-    template_file = StringIO('{"type_of_muffin": "{{ MUFFIN_TYPE }}"}')
+def test_render_json_template_valid(tmpdir):
+    template_path, template_name = _write_template_to_disk(
+        tmpdir, 'template.json',
+        '{"type_of_muffin": "{{ MUFFIN_TYPE }}"}',
+    )
 
-    rendered_template = render_json_template(template_file, **{"MUFFIN_TYPE": "banana"})
+    rendered_template = render_json_template(template_path, template_name, **{"MUFFIN_TYPE": "banana"})
     assert "type_of_muffin" in rendered_template
     assert rendered_template["type_of_muffin"] == "banana"
 
 
-def test_render_json_template_invalid_json_unquoted_string():
-    template_file = StringIO('{"type_of_muffin": {{ MUFFIN_TYPE }}}')
+def test_render_json_template_invalid_json_unquoted_string(tmpdir):
+    template_path, template_name = _write_template_to_disk(
+        tmpdir, 'template.json',
+        '{"type_of_muffin": {{ MUFFIN_TYPE }}}',
+    )
 
     with pytest.raises(InvalidJSONError):
-        render_json_template(template_file, **{"MUFFIN_TYPE": "banana"})
+        render_json_template(template_path, template_name, **{"MUFFIN_TYPE": "banana"})
 
 
-def test_render_json_template_invalid_json_missing_value():
-    template_file = StringIO('{"type_of_muffin": {{ MUFFIN_TYPE }}}')
+def test_render_json_template_invalid_json_missing_value(tmpdir):
+    template_path, template_name = _write_template_to_disk(
+        tmpdir, 'template.json',
+        '{"type_of_muffin": {{ MUFFIN_TYPE }}}',
+    )
 
     with pytest.raises(InvalidJSONError):
-        render_json_template(template_file, **{"MUFFIN_TYPE": ""})
+        render_json_template(template_path, template_name, **{"MUFFIN_TYPE": ""})
 
 
-def test_render_json_template_missing_value_raises():
-    template_file = StringIO('{"type_of_muffin": "{{ MUFFIN_TYPE }}"}')
+def test_render_json_template_missing_value_raises(tmpdir):
```

Comment:
Yeah, i see how this could be slightly confusing. Not a change that's relevant to this PR though so punting for now and will address with a more relevant test naming cleanup later.